cloudflare/src/framework/async_api.rs

```diff
@@ -55,20 +68,16 @@ impl Client {
             http_client,
         })
     }
-}
 
-#[async_trait]
-impl ApiClient for Client {
-    async fn request<ResultType, QueryType, BodyType>(
+    fn make_request<ResultType, QueryType, BodyType>(
         &self,
         endpoint: &(dyn Endpoint<ResultType, QueryType, BodyType> + Send + Sync),
-    ) -> ApiResponse<ResultType>
+    ) -> reqwest::RequestBuilder
```

Comment:
Making changes like this should be discussed before they become a PR. This changes the API of the library in a significant way.

Reply:
I agree! I was just hacking away at this to see if it was possible, and what the implications of a design like this would be. Then I figured, well, if I have the code and it compiles, may as well have the discussion on the PR instead of just throwing the code away. I'm fine if we don't want to merge this :)